engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -20,20 +21,49 @@
 static NSString* const kBackgroundFetchCapatibility = @"fetch";
 static NSString* const kRestorationStateAppModificationKey = @"mod-date";
 
+@interface FlutterSceneDelegate : NSObject <UIWindowSceneDelegate>
+@property(nonatomic, strong, nullable) UIWindow* window;
+@end
+
+@implementation FlutterSceneDelegate
+
+- (void)scene:(UIScene*)scene
+    willConnectToSession:(UISceneSession*)session
+                 options:(UISceneConnectionOptions*)connectionOptions {
+  NSObject<UIApplicationDelegate>* appDelegate = FlutterSharedApplication.application.delegate;
+  if (appDelegate.window.rootViewController) {
+    // If this is not nil we are running into a case where someone is manually
+    // performing root view controller setup outside of Storyboard initialization.
+    UIWindowScene* windowScene = (UIWindowScene*)scene;
+    self.window = [[UIWindow alloc] initWithWindowScene:windowScene];
+    self.window.rootViewController = appDelegate.window.rootViewController;
+    appDelegate.window = self.window;
+    [self.window makeKeyAndVisible];
+  }
+}
+
+@end
+
 @interface FlutterAppDelegate ()
 @property(nonatomic, copy) FlutterViewController* (^rootFlutterViewControllerGetter)(void);
 @property(nonatomic, strong) FlutterPluginAppLifeCycleDelegate* lifeCycleDelegate;
+@property(nonatomic, strong) FlutterLaunchEngine* launchEngine;
 @end
 
 @implementation FlutterAppDelegate
 
 - (instancetype)init {
   if (self = [super init]) {
     _lifeCycleDelegate = [[FlutterPluginAppLifeCycleDelegate alloc] init];
+    _launchEngine = [[FlutterLaunchEngine alloc] init];
   }
   return self;
 }
 
+- (nullable FlutterEngine*)grabLaunchEngine {
+  return [self.launchEngine grabEngine];
```

Comment:
it's a bit confusing that the name `grabLaunchEngine` returns the *engine* rather than the *launch engine*. 

How about renaming `FlutterLaunchEngine` as `FlutterEngineLaunchContainer`, and rename this `grabLaunchEngine` `grabEngineFromLaunchContainer`?